tests/src/python/test_qgsserver_accesscontrol_wms_getmap_postgres.py

```diff
@@ -0,0 +1,260 @@
+import urllib.request
+import urllib.parse
+import urllib.error
+from test_qgsserver_accesscontrol import TestQgsServerAccessControl
+from utilities import unitTestDataPath
+
+from qgis.core import QgsProviderRegistry
+from qgis.server import QgsAccessControlFilter
+
+
+class RestrictedAccessControlPG(QgsAccessControlFilter):
+
+    """ Used to have restriction access """
+
+    # Be able to deactivate the access control to have a reference point
+    _active = False
+
+    def __init__(self, server_iface):
+        super(QgsAccessControlFilter, self).__init__(server_iface)
+
+    def layerFilterExpression(self, layer):
+        """ Return an additional expression filter """
+
+        if not self._active:
+            return super(RestrictedAccessControlPG, self).layerFilterExpression(layer)
+
+        if layer.name() == "someData" or layer.name() == "someDataLong":
+            return "pk > 2"
+        else:
+            return None
+
+    def layerFilterSubsetString(self, layer):
+        """ Return an additional subset string (typically SQL) filter """
+
+        if not self._active:
+            return super(RestrictedAccessControlPG, self).layerFilterSubsetString(layer)
+
+        if layer.name() == "someData" or layer.name() == "someDataLong":
+            return "pk > 2"
+        else:
+            return None
+
+    def layerPermissions(self, layer):
+        """ Return the layer rights """
+
+        return super(RestrictedAccessControlPG, self).layerPermissions(layer)
+
+    def authorizedLayerAttributes(self, layer, attributes):
+        """ Return the authorised layer attributes """
+
+        return super(RestrictedAccessControlPG, self).authorizedLayerAttributes(layer, attributes)
+
+    def allowToEdit(self, layer, feature):
+        """ Are we authorise to modify the following geometry """
+
+        return super(RestrictedAccessControlPG, self).allowToEdit(layer, feature)
+
+    def cacheKey(self):
+        return "r" if self._active else "f"
+
+
+class TestQgsServerAccessControlWMSGetMapPG(TestQgsServerAccessControl):
+    """QGIS Server Access Control WMS Tests"""
+
+    @classmethod
+    def setUpClass(cls):
+
+        super().setUpClass()
+
+        if 'QGIS_PGTEST_DB' in os.environ:
+            cls.dbconn = os.environ['QGIS_PGTEST_DB']
+        else:
+            cls.dbconn = 'service=qgis_test dbname=qgis_test sslmode=disable '
+
+        # Test layer
+        md = QgsProviderRegistry.instance().providerMetadata('postgres')
+        uri = cls.dbconn + ' dbname=qgis_test sslmode=disable '
+        conn = md.createConnection(uri, {})
+        conn.executeSql('DROP TABLE IF EXISTS "qgis_test"."someDataLong" CASCADE')
+        conn.executeSql('SELECT * INTO "qgis_test"."someDataLong" FROM "qgis_test"."someData"')
+        conn.executeSql('ALTER TABLE "qgis_test"."someDataLong" ALTER COLUMN "pk" TYPE bigint')
+        conn.executeSql('ALTER TABLE "qgis_test"."someDataLong" ALTER COLUMN "pk" TYPE bigint')
```

Comment:
```suggestion
```

Reply:
It is the same in `test_qgsserver_accesscontrol_wms_getfeatureinfo_postgres.py`. I am removing it.

Reply:
Looks like a duplicated line to me, can you please fix the other file test_qgsserver_accesscontrol_wms_getfeatureinfo_postgres.py too? Thanks!

Reply:
I have fixed it, thanks to pointed it.